sklearn/model_selection/_split.py

```diff
@@ -1472,6 +1472,11 @@ def _iter_indices(self, X, y, groups=None):
         y = check_array(y, ensure_2d=False, dtype=None)
         n_train, n_test = _validate_shuffle_split(n_samples, self.test_size,
                                                   self.train_size)
+
+        if y.ndim == 2:
+            # for multi-label y, map each distinct row to its string repr:
+            y = np.array([str(row) for row in y])
```

Comment:
Can you try our cython `murmurhash`? Maybe it will be faster? + it's optimized for numpy arrays (if you would consider each row as one?)